lib/arch/generic/dec_tail.c

```diff
@@ -5,6 +5,7 @@
 		if ((q = base64_table_dec[*c++]) >= 254) {
 			st.eof = BASE64_EOF;
 			// Treat character '=' as invalid for byte 0:
+			ret = 0;
```

Comment:
Is this strictly necessary when `ret` defaults to 0? (I.e. we assume failure and must "prove" success.)